File: leetcode/119.pascals-triangle-ii.js
/*
 * @lc app=leetcode id=119 lang=javascript
 *
 * [119] Pascal's Triangle II
 */
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
  return pascalsTriangleRow(0, rowIndex, new Array(rowIndex + 1));

  function pascalsTriangleRow(i, n, arr) {
    if (i > n) return arr;

    for (let j = i; j >= 0; j--) {
      if (j === 0 || j === i) {
        arr[j] = 1;
      } else {
        arr[j] = arr[j - 1] + arr[j];
      }
    }

    return pascalsTriangleRow(i + 1, n, arr);
  }
};

File: leetcode/119.pascals-triangle-ii.java
/*
 * @lc app=leetcode id=119 lang=java
 *
 * [119] Pascal's Triangle II
 */

// @lc code=start
class Solution {
    public List<Integer> getRow(int rowIndex) {
        if (rowIndex == 0) {
            return Arrays.asList(1);
        }

        List<Integer> preList = getRow(rowIndex - 1);
        List<Integer> curList = new ArrayList<>();
        for (int i = 0; i < preList.size(); i++) {
            if (i == 0) {
                curList.add(1);
            }
            if (i > 0) {
                curList.add(preList.get(i - 1) + preList.get(i));
            }
            if (i == preList.size() - 1) {
                curList.add(1);
            }
        }

        return curList;
    }
}
// @lc code=end

